packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -91,6 +98,22 @@ abstract class ApplicationBase extends cdk.Resource implements IApplication {
     }
   }
 
+  /**
+   * Share application resource with target accounts.
+   * The application will become available to end users within targets.
+   * @param shareOptions
+   */
+  public shareResource(shareOptions: ShareOptions): void {
+    const principals = getPrincipalsforSharing(shareOptions);
+    const shareName = `RAMShare${hashValues(this.node.addr, ...principals)}`;
+    new CfnResourceShare(this, shareName, {
```

Comment:
We could consider since we have hash for resource name we can either throw an error if someone tries to add same share twice or ignore it, like we do in other SC resources.

Reply:
I will think about this one a bit more. I think some logic to ignore it may be sufficient.